egs/wsj/s5/utils/lang/make_subword_lexicon_fst.py

```diff
@@ -0,0 +1,240 @@
+            word = (word if i == 0 else constant_variable.eps)
+            cost = (pron_cost if i == 0 else 0.0)
+            print_arc(current_state, next_state, phones[i], word, cost)
+            current_state = next_state
+            next_state += 1
+    
+        i = len(phones) - 1
+        phone = (phones[i] if i >= 0 else constant_variable.eps)
+        word = (word if i <= 0 else constant_variable.eps)
+        cost = (pron_cost if i <= 0 else 0.0)
+        if to_state != constant_variable.sil_nonsil:
+            print_arc(current_state, to_state, phone, word, cost)
+        else:
+            p_cost = cost
+            cost = no_sil_cost + p_cost
+            print_arc(current_state, loop_state, phone, word, cost)
+            cost = sil_cost + p_cost
+            print_arc(current_state, sil_state, phone, word, cost)
+    
+        return next_state
+
+    sil_cost = -math.log(sil_prob)
+    no_sil_cost = -math.log(1 - sil_prob)
+
+    start_state = 0
+    loop_state = 1
+    sil_state = 2
+    prefix_state = 3
+    next_state = 4
+
+    print_arc(start_state, loop_state, constant_variable.eps, constant_variable.eps, no_sil_cost)
+    print_arc(start_state, sil_state, constant_variable.eps, constant_variable.eps, sil_cost)
+    if sil_disambig is None:
+        print_arc(sil_state, loop_state, sil_phone, constant_variable.eps, 0.0)
+    else:
+        disambig_state = next_state
+        next_state += 1
+        print_arc(sil_state, disambig_state, sil_phone, constant_variable.eps, 0.0)
+        print_arc(disambig_state, loop_state, sil_disambig, constant_variable.eps, cost=0.0)
+
+    for (word, pron_prob, phones) in lexicon:
+        cost = -math.log(pron_prob)
+
+        if is_end(word, separator):
+            if position_dependent:
+                position = get_position(phones, constant_variable)
+                next_state = print_word_with_silence(word, cost, phones, position, next_state)
+            else:
+                next_state = print_word_with_silence(word, cost, phones, constant_variable.end, next_state)
+                next_state = print_word_with_silence(word, cost, phones, constant_variable.word, next_state)
+        else:
+            # the word end with separator which should be followed by subword
+            if position_dependent:
+                position = get_position(phones, constant_variable)
+                next_state = print_word_with_silence(word, cost, phones, position, next_state)
+            else:
+                next_state = print_word_with_silence(word, cost, phones, constant_variable.begin, next_state)
+                next_state = print_word_with_silence(word, cost, phones, constant_variable.middle, next_state)
+
+    print("{state}\t{final_cost}".format(state=loop_state, final_cost=0.0))
+
+def main():
+    variable = namedtuple("variable", "eps begin middle end word sil_nonsil")
+    constant_variable = variable(eps="<eps>", begin="begin", middle="middle", 
```

Comment:
constant_variable is a bad name (`constant` and `variable` are antonyms).  It looks like what you really need here is an Enum.  Look it up.

You also need much more documentation.  You don't even attempt to explain what these things mean and what the purpose of them is.